js/core/src/registry.ts

```diff
@@ -53,6 +55,15 @@ function pluginsByName(): Record<string, PluginProvider> {
   }
   return global[PLUGINS_BY_NAME];
 }
+function schemasByName(): Record<
+  string,
+  { schema?: z.ZodTypeAny; jsonSchema?: JSONSchema }
+> {
+  if (global[SCHEMAS_BY_NAME] === undefined) {
+    global[SCHEMAS_BY_NAME] = {};
+  }
+  return global[SCHEMAS_BY_NAME];
+}
```

Comment:
Do we ever want/need to access the schema from the dev ui (to display/inspect)?  or is the idea that the schema refs are always going to be fully resolved in dotprompts, etc. before the action is exposed to the dev ui?

Reply:
Good question, I _think_ they'll be resolved in the prompt loading process but there's probably some confirming of that needs to happen.

Reply:
Looking at the unit test it looks like it gets resolved... should be fine I think.